src/main/java/ro/msg/learning/shop/domain/OrderDetail.java

```diff
@@ -0,0 +1,32 @@
+package ro.msg.learning.shop.domain;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.JdbcTypeCode;
+import org.hibernate.type.SqlTypes;
+import ro.msg.learning.shop.domain.primarykeys.OrderDetailKey;
+
+@Entity
+@Builder
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Table(name = "order_detail")
+@IdClass(OrderDetailKey.class)
+public class OrderDetail {
+    @Id
+    @JdbcTypeCode(SqlTypes.UUID)
+    @Column(name = "order_id")
+    private Order orderId;
```

Comment:
The name of the field should be `order`, not `orderId` as the field includes the whole order object. You might need to adjust the `OrderDetailKey` to be able to change the naming here (there it is ok if you write `private UUID order`). 😄 

Please adjust the other fields too (e.g., below, write `private Location location`).